### PR TITLE
[PAN-1866] add python shared CI

### PR DESCRIPTION
**Pull request by danut13**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## Description

- Implement Pantos Python CI shared pipeline
- Split the actions which install Poetry from the action which install the dependency of the project: this is done because sometimes we need **only** poetry to build wheels and we should reuse this step and not redefine it in individual projects.
- Removed some inputs from the previous version: They were actually redundant and we can achieve the same behaviour without them.
- Compared to the previous shared pipeline that we had, I split the SonarCloud step from the unit test one. They should be independent from my point of view, especially because this step will fail if the pipeline is triggered from forks, and it should not have an impact on the unit test step. This was accomplished by passing the coverage artifacts to the SonarCloud responsible job.
- Tagged it to v1 and v1.0.0 according to https://github.com/actions/toolkit/blob/main/docs/action-versioning.md

## QA Instructions

- Pantos Python projects should be able to call the reusable workflow to check the desired properties for CI
- Pantos Python projects should be able to upload artifacts to SonarCloud using the workflow (only after the CI is successfully finished)

### CODE REVIEW

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      - uses: actions/download-artifact@v4
+        if: ${{ inputs.test-database }}
+        with:
+          name: code-coverage-postgres-report
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+            GITHUB_TOKEN:  ${{ secrets.GH_TOKEN || github.token }}  # Needed to get PR information, if any
+            SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (juanmanuel-tirado) [inline .github/workflows/python-ci.yml]: New line at the end of the file

**Review by juanmanuel-tirado — DISMISSED** [review]
LG2M

File: .github/actions/install-poetry/action.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: 'Install dependencies'
+description: 'Install Python library dependencies using Poetry'
```
Comment (chandraprakash) [inline .github/actions/install-poetry/action.yml]: 'Set up Python and install Poetry'

File: .github/workflows/python-ci.yml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+            token: ${{ secrets.GH_TOKEN || github.token }}
+
+      - uses: pantos-io/ci-workflows/.github/actions/install-python-deps@v1
+      
+      - name: UnitTest
+        run: |
+            make coverage
```
Comment (juanmanuel-tirado) [inline .github/workflows/python-ci.yml:116]: Are not `make coverage` and `poetry run coverage xml` doing the same?

File: .github/actions/install-poetry/action.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+runs:
+  using: composite
+  steps:
+    - name: Set up python
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install Poetry
```
Comment (chandraprakash) [inline .github/actions/install-poetry/action.yml:19]: may be worth caching this too? 
[cached-poetry-dependencies](https://github.com/snok/install-poetry?tab=readme-ov-file#caching-the-poetry-installation)

File: .github/workflows/python-ci.yml (hunk 3 above)
Comment (danut13) [reply .github/workflows/python-ci.yml:116]: Nope, `make coverage` is running the unit tests and generates the coverage report in a `.coverage` which is not understood by SonarCloud. `poetry run coverage xml` converts that file in a xml format which can then be uploaded to SonarCloud

File: .github/actions/install-poetry/action.yml (hunk 4 above)
Comment (danut13) [reply .github/actions/install-poetry/action.yml:19]: Good one, I added it

File: .github/actions/install-poetry/action.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  using: composite
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        
+    - name: Load cached Poetry installation
+      id: cached-poetry
+      uses: actions/cache@v3
```
Comment (chandraprakash) [inline .github/actions/install-poetry/action.yml]: will be good to use actions/cache@v4 if possible. No issues if can't use it.

**Review by chandraprakash — DISMISSED** [review]

**Review by chandraprakash — APPROVED** [review]

**Review by juanmanuel-tirado — APPROVED** [review]
LG2M

**Pull request merged**